src/components/TodoContainer.js

```diff
@@ -1,140 +1,84 @@
 import TodoItem from "./TodoItem";
-import styled from "styled-components";
-
-const TodoWrapper = styled.div`
-  font-family: "ubuntu";
-  margin: 30px auto;
-  width: 480px;
-  border: 3px solid #f5f5f5;
-  border-radius: 5px;
-  padding: 30px;
-  text-align: center;
-  box-shadow: 3px 3px 5px #ccc;
-`;
-
-const Title = styled.h1`
-  color: #28262c;
-  font-size: 48px;
-`;
-
-const CreateTodo = styled.div`
-  margin: 20px auto;
-`;
-
-const TodoInput = styled.input`
-  margin-right: 10px;
-  width: 300px;
-  height: 24px;
-  border: 1px solid #ccc;
-  border-radius: 5px;
-  box-shadow: 1px 1px 3px #ccc;
-`;
-
-const AddButton = styled.button`
-  font-family: "ubuntu";
-  width: 80px;
-  background: black;
-  color: white;
-  border-radius: 3px;
-  box-shadow: 1px 1px 3px #666;
-  border: none;
-  padding: 5px;
-`;
-
-const SelectTodo = styled.div``;
-
-const AllButton = styled.button`
-  font-family: "ubuntu";
-  width: 80px;
-  background: #39393a;
-  color: white;
-  border-radius: 3px;
-  box-shadow: 1px 1px 3px #666;
-  border: none;
-  padding: 5px;
-`;
-
-const ActiveButton = styled.button`
-  font-family: "ubuntu";
-  width: 80px;
-  background: #ffe74c;
-  color: #333;
-  border-radius: 3px;
-  box-shadow: 1px 1px 3px #666;
-  border: none;
-  padding: 5px;
-  margin: 0 10px;
-`;
-
-const CompletedButton = styled.button`
-  font-family: "ubuntu";
-  width: 80px;
-  background: #666;
-  color: white;
-  border-radius: 3px;
-  box-shadow: 1px 1px 3px #666;
-  border: none;
-  padding: 5px;
-`;
-
-const TodoList = styled.div`
-  margin-top: 10px;
-`;
-
-const ClearTodo = styled.button`
-  font-family: "ubuntu";
-  width: 120px;
-  background: #c8553d;
-  color: white;
-  border-radius: 3px;
-  box-shadow: 1px 1px 3px #666;
-  border: none;
-  padding: 5px;
-  margin: 10px;
-`;
+import {
+  TodoWrapper,
+  Title,
+  CreateTodo,
+  TodoInput,
+  AllButton,
+  ActiveButton,
+  CompletedButton,
+  TodoList,
+  ClearTodo,
+} from "./styles/TodoContainer.style";
 
 export default function TodoContainer() {
   const id = useRef(1);
   const [todos, setTodos] = useState([]);
   const [value, setValue] = useState("");
   const [filter, setFilter] = useState("all");
 
-  const handleAddTodo = useCallback(() => {
-    if (!value) return alert("wanna type something?");
-    setTodos((todo) => [{ id: id.current, content: value }, ...todos]);
-    setValue("");
-    id.current++;
-  }, [todos, value]);
+  useEffect(() => {
+    if (localStorage.getItem("savedLocalTodos")) {
+      const savedLocalTodos = JSON.parse(
+        localStorage.getItem("savedLocalTodos")
+      );
+      setTodos(savedLocalTodos);
+    }
+  }, []);
+
+  const handleAddTodo = useCallback(
+    (e) => {
+      if (!value || value.trim() === "") return;
+      if (e.key === "Enter") {
+        const todo = { id: id.current, content: value, isDone: false };
+        localStorage.setItem(
+          "savedLocalTodos",
+          JSON.stringify([...todos, todo])
+        );
```

Comment:
Instead of directly using localStorage, can you create a state for the items changes, then update to localstorage when the state changes? (adding hook, check this https://www.freecodecamp.org/news/how-to-use-localstorage-with-react-hooks-to-set-and-get-items/ )